xmodule/tests/test_item_bank.py

```diff
@@ -0,0 +1,498 @@
+        self.item_bank = self.make_block(
+            category="itembank", parent_block=self.vertical, max_count=1, display_name="My Item Bank", publish_item=True
+        )
+        self.items = [
+            self.make_block(
+                category="problem", parent_block=self.item_bank, display_name=f"My Item {i}",
+                data=f"<p>Hello world from problem {i}</p>",
+                publish_item=True,
+            )
+            for i in range(4)
+        ]
+        self.publisher = Mock()  # for tests that look at analytics
+        self._reload_item_bank()
+
+    def _bind_course_block(self, block):
+        """
+        Bind a block (part of self.course) so we can access student-specific data.
+
+        (Not clear if this is necessary since XModules are all removed now. It's possible that this
+         could be removed without breaking tests.)
+        """
+        prepare_block_runtime(block.runtime, course_id=block.location.course_key)
+
+        def get_block(descriptor):
+            """Mocks module_system get_block function"""
+            prepare_block_runtime(descriptor.runtime, course_id=block.location.course_key)
+            descriptor.runtime.get_block_for_descriptor = get_block
+            descriptor.bind_for_student(self.user_id)
+            return descriptor
+
+        block.runtime.get_block_for_descriptor = get_block
+
+    def _reload_item_bank(self):
+        """
+        Reload self.item_bank. Do this if you want its `.children` list to be updated with what's in the db.
+
+        HACK: These test cases don't persist student state, but some tests need `selected` to persist betweem item_bank
+        reloads. So, we just transfer it from the old item_bank object instance to the new one, as if it were persisted.
+        """
+        selected = self.item_bank.selected
+        self.item_bank = self.store.get_item(self.item_bank.location)
+        self._bind_course_block(self.item_bank)
+        if selected:
+            self.item_bank.selected = selected
+        self.item_bank.runtime.publish = self.publisher
+
+
+@skip_unless_cms
+class TestItemBankForCms(ItemBankTestBase):
+    """
+    Test Studio ItemBank behaviors -- export/import, validation, author-facing views.
+    """
+    def test_xml_export_import_cycle(self):
+        """
+        Test the export-import cycle.
+        """
+        # Export self.item_bank to the virtual filesystem
+        export_fs = MemoryFS()
+        self.item_bank.runtime.export_fs = export_fs  # pylint: disable=protected-access
+        node = etree.Element("unknown_root")
+        self.item_bank.add_xml_to_node(node)
+
+        # Read back the itembank OLX
+        with export_fs.open('{dir}/{file_name}.xml'.format(
+            dir=self.item_bank.scope_ids.usage_id.block_type,
+            file_name=self.item_bank.scope_ids.usage_id.block_id
+        )) as f:
+            actual_olx_export = f.read()
```

Comment:
I don't like how this relies weird export behavior coming from `XmlMixin`, such that `add_xml_to_node` has a side-effect of writing definitions to the memory filesystem. According to the XmlMixin docstring, this behavior [is "specifically for `customtag` ?"](https://github.com/kdmccormick/edx-platform/blob/74af0e71e840af07f0085bb510dec6b4ef4616a1/xmodule/xml_block.py#L413-L420).

I was going to say, I'd rather you remove `XmlMixin` and just use `actual_olx_export = etree.tostring(node, encoding="unicode", pretty_print=True)` to get the string version, instead of encoding these export nuances into the test case. If you need more advanced serialization (with children and static assets), call [`serialize_xblock_to_olx(self.item_bank)`](https://github.com/openedx/edx-platform/blob/e47fdbf95a7aba58e9be6ba9473a269d6cf3f79b/openedx/core/lib/xblock_serializer/api.py#L8) instead.

BUT I guess you're also relying on the fs-serialization of `problem` blocks and you're wanting to test code that's as similar as possible to the actual export-import cycle here? I wish our import code wasn't so messy.

Reply:
I can try to simplify it. 

> BUT I guess you're also relying on the fs-serialization of problem blocks and you're wanting to test code that's as similar as possible to the actual export-import cycle here? 

yes, but also `test_library_content_block.py` still exists and has the near-identical test case that I cribbed this one from, so I think it's safe to use a cleaner version of the test case in `test_item_bank.py`

Reply:
OK. Don't spend a ton of time on it though; more important we get this merged ASAP.

Reply:
OK, I'd rather do this in a follow-up then